caddyhttp/proxy/reverseproxy.go

```diff
@@ -254,6 +254,10 @@ func (rp *ReverseProxy) copyResponse(dst io.Writer, src io.Reader) {
 
 func copyHeader(dst, src http.Header) {
 	for k, vv := range src {
+		// some predefined headers (e.g. MIME plugin), skip.
+		if _, ok := dst[k]; ok {
+			continue
+		}
```

Comment:
And then, according to [this tweet](https://twitter.com/adamwathan/status/779018371090178053), there's the docs for XSendfile: https://www.nginx.com/resources/wiki/start/topics/examples/xsendfile/

Which apparently doesn't modify these headers:

> Content-Type
Content-Disposition
Accept-Ranges
Set-Cookie
Cache-Control
Expires

However, this seems like it has more to do with the `internal` directive than proxy. Where do you think is the best place to implement this fix?